lms/djangoapps/instructor/tests/test_enrollment.py

```diff
@@ -494,6 +499,81 @@ def test_delete_submission_scores_attempts_children(self):
         self.assertEqual(unrelated_state['brains'], 'zombie')
 
 
+class TestStudentModuleGrading(SharedModuleStoreTestCase):
+    """
+    Tests the effects of student module manipulations
+    on student grades.
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(TestStudentModuleGrading, cls).setUpClass()
+        cls.course = CourseFactory.create()
+        cls.course_key = cls.course.location.course_key
+        cls.chapter = ItemFactory.create(
+            parent=cls.course,
+            category="chapter",
+            display_name="Test Chapter"
+        )
+        cls.sequence = ItemFactory.create(
+            parent=cls.chapter,
+            category='sequential',
+            display_name="Test Sequential 1",
+            graded=True
+        )
+        cls.vertical = ItemFactory.create(
+            parent=cls.sequence,
+            category='vertical',
+            display_name='Test Vertical 1'
+        )
+        problem_xml = MultipleChoiceResponseXMLFactory().build_xml(
+            question_text='The correct answer is Choice 3',
+            choices=[False, False, True, False],
+            choice_names=['choice_0', 'choice_1', 'choice_2', 'choice_3']
+        )
+        cls.problem = ItemFactory.create(
+            parent=cls.vertical,
+            category="problem",
+            display_name="Test Problem",
+            data=problem_xml
+        )
+        cls.request = get_request_for_user(UserFactory())
+        cls.user = cls.request.user
+
+    def _get_subsection_grade_and_verify(self, all_earned, graded_earned, all_possible, graded_possible):
```

Comment:
Nit: reorder the parameters, so it's easier to understand the fractions:
`all_earned, all_possible, graded_earned, graded_possible`